pkg/rancher-desktop/main/snapshots/snapshots.ts

```diff
@@ -0,0 +1,89 @@
+import { spawnFile } from '@pkg/utils/childProcess';
+import Logging from '@pkg/utils/logging';
+import { getRdctlPath } from '@pkg/utils/paths';
+
+import { Snapshot, SpawnResult } from './types';
+
+const console = Logging.snapshots;
+
+function parseFields(line: string): string[] {
+  return line?.split(/\s{2,}/) || [];
+}
+
+function parseLines(line: string): string[] {
+  return line?.split(/\r?\n/) || [];
+}
+
+function parseRows(response: SpawnResult) {
+  const data = parseLines(response.stdout).filter(line => line);
+  const [header, ...lines] = data;
+
+  const fields = parseFields(header);
+
+  return lines.map(line => fields.reduce((acc, f, index) => ({
+    ...acc,
+    [f.toLowerCase()]: parseFields(line)[index],
+  }), {} as Snapshot));
+}
+
+class SnapshotsError {
+  readonly isSnapshotError = true;
+  message: string;
+
+  constructor(stderr: string) {
+    this.message = parseLines(stderr)[0];
+  }
+}
+
+class SnapshotsImpl {
+  private async rdctl(commandArgs: string[]): Promise<SpawnResult> {
+    try {
+      const rdctlPath = getRdctlPath();
+
+      return await spawnFile(rdctlPath || '', commandArgs, { stdio: ['ignore', 'pipe', 'pipe'] });
+    } catch (err: any) {
+      return {
+        stdout: err?.stdout ?? '', stderr: err?.stderr ?? '', error: err,
+      };
+    }
+  }
+
+  async list(): Promise<Snapshot[]> {
+    const response = await this.rdctl(['snapshot', 'list']);
+
+    if (response.stderr) {
```

Comment:
Did you want to test `if (response.error)` instead? Why are we keying based off of stderr having output?

Same with the other methods.

Reply:
Agree, checking `response.error` is more appropriate.